### PR TITLE
Add all in one comment

### PR DESCRIPTION
**Pull request by anyasabo**
It at least wasn't clear to me that `make generate-all-in-one` needed me to give it the version. I also wonder if we should just not set a default here and instead error out if the target is called but it isn't specified.

### CODE REVIEW

**Comment by thbkrkr** [comment]
The `LATEST_RELEASED_IMG` variable has been temporarily introduced when there was no release process. But since then, we have built things over. :building_construction: 

So, I'm +1 to add this comment as a transient palative care.

> I also wonder if we should just not set a default here and instead error out if the target is called but it isn't specified.

Yes, I think we should do that or even see how to completely remove it.

File: Makefile [hunk 1]
```diff
@@ -19,7 +19,7 @@ NAME       	?= eck-operator
 VERSION    	?= $(shell cat VERSION)
 SNAPSHOT   	?= true
 
-LATEST_RELEASED_IMG ?= "docker.elastic.co/eck/$(NAME):0.8.0"
+LATEST_RELEASED_IMG ?= "docker.elastic.co/eck/$(NAME):0.9.0"
```
Comment (sebgl) [inline Makefile]: Would `ECK_IMAGE` be a better name?

**Comment by anyasabo** [comment]
So one issue I found with requiring the variable to be set explicitly is that we generate the all in one as part of `generate`: https://github.com/elastic/cloud-on-k8s/blob/master/Makefile#L116

I wonder if we should remove that? I don't have all of the background on why this was added so definitely open to suggestions

**Comment by david-kow** [comment]
@anyasabo looking at the failure, LATEST_RELEASED_IMG has to be set for PR job.

**Comment by pebrc** [comment]
@anyasabo this PR seems to have stalled. Either we fix the PR job variable or we close this, wdyt?

**Comment by anyasabo** [comment]
I don't have a good idea on how to fix this immediately so won't be able to get to it soon. Maybe I should revert to just the comment and open an issue to dive into in the future?

**Comment by thbkrkr** [comment]
I updated the PR to remove `LATEST_RELEASED_IMG`.

`LATEST_RELEASED_IMG` was used to set `OPERATOR_IMAGE` before calling `$(MAKE) --no-print-directory -sC config/operator generate-all-in-one` and was always set to almost the same value as `OPERATOR_IMAGE`. The only difference was the presence of the `push.` prefix used when the operator is pushed in the Elastic Docker repo (in the nightly and release jobs). So, it's pretty safe to remove it.

I also removed `USE_ELASTIC_DOCKER_REGISTRY` which was set to true in the nightly and release jobs to add the `docker login` step before pushing the image and updated the condition to detect when this step is used. Finally, I reused this condition to know if the `push.` prefix is required.

I think there are other simplifications that can be done but we should proceed step by step to avoid a big-bang.

**Pull request merged**